File: front/src/app/models/fiche-emargement.ts
import {User} from "./user";

export class FicheEmargement {
  public id: string;
  public nomCours: string;
  // public joinDate: Date;
  public listeEtudiantSigne: User[];


  constructor() {
    this.nomCours = '';
    this.listeEtudiantSigne = [];

  }

}

File: back/src/main/java/com/example/projetsem2qrcode/modele/FicheEmargement.java
package com.example.projetsem2qrcode.modele;


import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document(collection = "fiche_emargements")
@Data
public class FicheEmargement {

    @Id
    public String id;
    public String nomCours;
    public List<User> listeEtudiantSigne;

    public FicheEmargement(String nomCours) {
        this.nomCours = nomCours;
    }
}
